azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -947,17 +960,18 @@ def _run_op(self, op):
                         self.name, op.name, self.state
                     )
                 )
+                self.state = ReconnectState.LOGICALLY_DISCONNECTED
                 self._clear_reconnect_timer()
                 self._complete_waiting_connect_ops(
                     pipeline_exceptions.OperationCancelled("Explicit disconnect invoked")
```

Comment:
>pipeline_exceptions.OperationCancelled("Explicit disconnect invoked") [](start = 20, length = 69)

Unrelated to the rest of this PR I guess, but I don't get what's going on here, and why this gets set as a waiting connect op

Reply:
Not sure if you selected the wrong line here.  This code (DisconnectOperation, WAITING_TO_RECONNECT) covers the case where the connection was dropped and it's waiting to retry when the user calls disconnect().  In that case, any operations that were waiting for the reconnect to succeed need to be cancelled.  By calling disconnect(), the user was effectively cancelling all operations that were pending. 

---
In reply to: [439686461](https://github.com/Azure/azure-iot-sdk-python/pull/576#discussion_r439686461) [](ancestors = 439686461)

Reply:
I'm not sure because the line you selected is the one that completes (cancels) all of the waiting connection ops.

---
In reply to: [440422365](https://github.com/Azure/azure-iot-sdk-python/pull/576#discussion_r440422365) [](ancestors = 440422365,439686461)